plugins/workspace-minimap/src/focus_mode.ts

```diff
@@ -0,0 +1,186 @@
+          Blockly.utils.Svg.RECT, {
+            'x': 0,
+            'y': 0,
+            'rx': borderRadius,
+            'ry': borderRadius,
+            'fill': 'black',
+          }, mask);
+
+      // Theme.
+      this.background.setAttribute('fill', '#e6e6e6');
+
+      // Add the svg group to the minimap.
+      const parentSvg = this.minimapWorkspace.getParentSvg();
+      if (parentSvg.firstChild) {
+        parentSvg.insertBefore(this.svgGroup, parentSvg.firstChild);
+      } else {
+        parentSvg.appendChild(this.svgGroup);
+      }
+
+      window.addEventListener('resize', () => void this.update());
+      this.onChangeWrapper = this.onChange.bind(this);
+      this.primaryWorkspace.addChangeListener(this.onChangeWrapper);
+
+      this.update();
+      this.initialized = true;
+    }
+
+
+    /**
+     * Disposes of the focus mode.
+     * Unlinks from all DOM elements and remove all event listeners
+     * to prevent memory leaks.
+     */
+    dispose() {
+      if (this.onChangeWrapper) {
+        this.primaryWorkspace.removeChangeListener(this.onChangeWrapper);
+      }
+      if (this.svgGroup) {
+        Blockly.utils.dom.removeNode(this.svgGroup);
+      }
+      this.svgGroup = null;
+      this.rect = null;
+      this.background = null;
+      this.initialized = false;
+    }
+
+
+    /**
+     * Handles events triggered on the primary workspace.
+     * @param event The event.
+     */
+    private onChange(event: Blockly.Events.Abstract): void {
+      if (blockEvents.has(event.type)) {
+        this.update();
+      }
+    }
+
+
+    /**
+     * Positions and sizes the highlight on the minimap
+     * based on the primary workspace.
+     */
+    private update(): void {
+      // Get the metrics.
+      const primaryMetrics = this.primaryWorkspace.getMetricsManager();
+      const minimapMetrics = this.minimapWorkspace.getMetricsManager();
+
+      const primaryView = primaryMetrics.getViewMetrics();
+      const primaryContent = primaryMetrics.getContentMetrics();
+      const minimapContent = minimapMetrics.getContentMetrics();
+      const minimapSvg = minimapMetrics.getSvgMetrics();
+
+      // Get the workscape to pixel scale on the minimap.
+      const scale = minimapContent.width /
+        minimapMetrics.getContentMetrics(true).width;
+
+      // Get the viewport size on a minimap scale.
+      const width = primaryView.width * scale;
+      const height = primaryView.height * scale;
+
+      // Get the viewport position in relation to the content.
+      let left = (primaryView.left - primaryContent.left) * scale;
+      let top = (primaryView.top - primaryContent.top) * scale;
+
+      // Account for the padding outside the content on the minimap.
+      left += (minimapSvg.width - minimapContent.width) / 2;
+      top += (minimapSvg.height - minimapContent.height) / 2;
+
+      // Set the svg attributes.
+      this.rect.setAttribute('transform',
+          'translate(' + left + ',' + top + ')');
```

Comment:
This might be a little more readable with backtick syntax: \`translate(${left}, ${top})\`

Reply:
Agreed